src/org/labkey/test/components/experiment/NodeDetail.java

```diff
@@ -0,0 +1,120 @@
+package org.labkey.test.components.experiment;
+
+import org.labkey.test.Locator;
+import org.labkey.test.components.Component;
+import org.labkey.test.components.WebDriverComponent;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+/**
+ * Automates the labkey UI for node details (implemented in /lineage/LineageNodeList.tsx)
+ */
+public class NodeDetail extends WebDriverComponent<NodeDetail.ElementCache>
+{
+    final WebElement _el;
+    final WebDriver _driver;
+
+    public NodeDetail(WebElement element, WebDriver driver)
+    {
+        _el = element;
+        _driver = driver;
+    }
+
+    public String getName()
+    {
+       return elementCache().nameElement.getText();
+    }
+
+    private boolean isNameLinked()
+    {
+        return elementCache().nameSpan.existsIn(this);
+    }
+
+    public void clickOverViewLink(boolean wait)
+    {
+        getWrapper().mouseOver(getComponentElement());
+        getWrapper().waitFor(()-> elementCache().overviewLink.isEnabled(), 1000);
+        if (wait)
+            getWrapper().clickAndWait(elementCache().overviewLink);
+        else
+            elementCache().overviewLink.click();
+    }
+
+    public void clickLineageGraphLink()
+    {
+        getWrapper().mouseOver(getComponentElement());
+        getWrapper().waitFor(()-> elementCache().lineageGraphLink.isEnabled(), 1000);
+        getWrapper().clickAndWait(elementCache().lineageGraphLink);
```

Comment:
I think you will need to add a wait parameter to this function call as well. You cannot depend upon clicking lineage to open/load a new page.

Reply:
good call.